### PR TITLE
Add semester controller

### PR DESCRIPTION
**Pull request by BrainMaestro**
Part of #62

### CODE REVIEW

**Comment by BrainMaestro** [comment]
Approved

**Comment by tenorz007** [comment]
First of all, there are too many file changes in this request. The name of this feature is add-semester-controller, yet I can see changes in transformer, repository, tests. You should know better than that and streamline the branch. If you need to make other changes make a new feature first for those changes and not jumble them all into one. @BrainMaestro

**Comment by BrainMaestro** [comment]
It is all related

**Comment by tenorz007** [comment]
If we were to think like that, I could create the route repository
migration and everything at once and say "it's all related" the whole
system is "all related"

**Comment by BrainMaestro** [comment]
I needed to touch the transformers to complete the controller function. Same with the repository. The only slightly unnecessary changes is a new line in 3 tests.

**Comment by tenorz007** [comment]
I didn't say they were unnecessary, you are supposed to separate those
things into separate features and you know that.

**Comment by BrainMaestro** [comment]
Yeah oversight

File: app/Http/Controllers/SemesterController.php [hunk 1]
```diff
@@ -2,59 +2,158 @@
...
 
-    public function update($id)
+    /**
+     * Add a question set to the semester.
+     *
+     * @param SemesterQuestionSetRequest $request
+     * @param $id
+     * @return mixed
+     */
+    public function addQuestionSet(SemesterQuestionSetRequest $request, $id)
```
Comment (tenorz007) [inline app/Http/Controllers/SemesterController.php]: This should be storeQuestionSet

File: app/Transformers/SemesterTransformer.php [hunk 2]
```diff
@@ -19,11 +20,14 @@ class SemesterTransformer extends TransformerAbstract
      */
     public function transform(Semester $semester)
     {
+        $repository = app(SemesterRepository::class);
```
Comment (tenorz007) [inline app/Transformers/SemesterTransformer.php]: You can't do this in a transformer

File: app/Http/Controllers/SemesterController.php (hunk 1 above)
Comment (BrainMaestro) [inline app/Http/Controllers/SemesterController.php]: Well that would be if it was saving a question set. But it's saving the relationship. So it's actually adding a new one to the semester. So I think the current name fits.


Sent from Outlook Mobile






On Wed, Mar 9, 2016 at 10:36 AM -0800, "tenorz007" <notifications@github.com> wrote:





>          }
>      }
>
> -    public function update($id)
> +    /**
> +     * Add a question set to the semester.
> +     *
> +     * @param SemesterQuestionSetRequest $request
> +     * @param $id
> +     * @return mixed
> +     */
> +    public function addQuestionSet(SemesterQuestionSetRequest $request, $id)

This should be storeQuestionSet

---
Reply to this email directly or view it on GitHub:
https://github.com/BrainMaestro/FCE/pull/80/files#r55565499

File: app/Transformers/SemesterTransformer.php (hunk 2 above)
Comment (BrainMaestro) [inline app/Transformers/SemesterTransformer.php]: Yeah it's not ideal, but no one says you can't do that. It's just helping me get the data. If you have a better way, I'll happily change it.


Sent from Outlook Mobile






On Wed, Mar 9, 2016 at 10:36 AM -0800, "tenorz007" <notifications@github.com> wrote:





> @@ -19,11 +20,14 @@ class SemesterTransformer extends TransformerAbstract
>       */
>      public function transform(Semester $semester)
>      {
> +        $repository = app(SemesterRepository::class);

You can't do this in a transformer

---
Reply to this email directly or view it on GitHub:
https://github.com/BrainMaestro/FCE/pull/80/files#r55565659

**Pull request closed without merging**